runtime/platform_isolate_manager.cc

```diff
@@ -0,0 +1,62 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/runtime/platform_isolate_manager.h"
+
+#include "flutter/runtime/dart_isolate.h"
+
+namespace flutter {
+
+bool PlatformIsolateManager::IsShutdown() {
+  std::scoped_lock lock(lock_);
+  return is_shutdown_;
+}
+
+bool PlatformIsolateManager::RegisterPlatformIsolate(Dart_Isolate isolate) {
+  std::scoped_lock lock(lock_);
+  if (is_shutdown_) {
+    // It's possible shutdown occured while we were trying to aquire the lock.
+    return false;
+  }
+  FML_DCHECK(platform_isolates_.find(isolate) == platform_isolates_.end());
+  platform_isolates_.insert(isolate);
+  return true;
+}
+
+void PlatformIsolateManager::RemovePlatformIsolate(Dart_Isolate isolate) {
+  // This method is only called by DartIsolate::OnShutdownCallback() during
+  // isolate shutdown. This can happen either during the ordinary platform
+  // isolate shutdown, or during ShutdownPlatformIsolates(). In either case
+  // we're on the platform thread.
+  // TODO(flutter/flutter#136314): Assert that we're on the platform thread.
+  // Need a method that works for ShutdownPlatformIsolates() too.
+  std::scoped_lock lock(lock_);
+  if (is_shutdown_) {
+    // Removal during ShutdownPlatformIsolates. Ignore, to avoid modifying
```

Comment:
You could assert the set is empty if you made the `ShutdownPlatformIsolates()` use `std::move()` the set out of the field & make field empty, before iterating.